source/neuropod/tests/test_ope_multiple_instances.cc

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2020 UATC, LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "neuropod/neuropod.hh"
+#include "neuropod/tests/ope_overrides.hh"
+
+#include <thread>
+
+TEST(test_ope_multiple_instances, multithreaded)
+{
+    const auto NUM_INSTANCES = 4;
+    const auto NUM_REQUESTS  = 2048;
+
+    // Create models
+    std::vector<neuropod::Neuropod> models;
+    for (unsigned long i = 0; i < NUM_INSTANCES; ++i)
+    {
+        neuropod::RuntimeOptions opts;
+        opts.use_ope = true;
+        models.emplace_back(
+            "neuropod/tests/test_data/pytorch_strings_model/", detail::ope_backend_location_overrides, opts);
+    }
+
+    // Start worker threads to run inference
+    std::vector<std::thread> workers;
+    for (size_t idx = 0; idx < models.size(); ++idx)
+    {
```

Comment:
I don't remember if this is necessary in C++ but definitely a problem in Go's goroutine and other languages: 

Copy idx to local var before sending it to closure otherwise it may happen that all threads runs with latest idx. I'd recommend to use this trick to avoid confusion (or comment please if you 100% sure that this is not a problem here)

 auto modelIndex = idx;

Reply:
This is not an issue here. The lambda captures idx by value so it is making a copy.